src/etrans/gpu/internal/espnorm_ctl_mod.F90

```diff
@@ -0,0 +1,64 @@
+MODULE ESPNORM_CTL_MOD
+CONTAINS
+SUBROUTINE ESPNORM_CTL(PSPEC,KFLD,KFLD_G,KVSET,KMASTER,PMET,PNORM)
+
+USE PARKIND1  ,ONLY : JPIM     ,JPRB
+USE YOMHOOK   ,ONLY : LHOOK,   DR_HOOK, JPHOOK, JPHOOK
+
+USE TPM_DIM         ,ONLY : R
+USE TPM_DISTR       ,ONLY : D, MYSETV, MYPROC
+
+USE ESPNORMD_MOD    ,ONLY : ESPNORMD
+USE SPNORMC_MOD     ,ONLY : SPNORMC
+
+USE TPMALD_DIM      ,ONLY : RALD
+!
+
+IMPLICIT NONE
+
+REAL(KIND=JPRB)    ,OPTIONAL, INTENT(IN)  :: PSPEC(:,:)
+INTEGER(KIND=JPIM) ,OPTIONAL, INTENT(IN)  :: KVSET(:)
+INTEGER(KIND=JPIM) ,OPTIONAL, INTENT(IN)  :: KMASTER
+REAL(KIND=JPRB)    ,OPTIONAL, INTENT(IN)  :: PMET(:)
+REAL(KIND=JPRB)    ,OPTIONAL, INTENT(OUT) :: PNORM(:)
+INTEGER(KIND=JPIM)          , INTENT(IN)  :: KFLD,KFLD_G
+INTEGER(KIND=JPIM) :: IVSET(KFLD_G)
+
+REAL(KIND=JPRB)    :: ZMET(0:R%NSPEC_G)
+
+REAL(KIND=JPRB)    :: ZSM(KFLD,D%NUMP)
+
+REAL(KIND=JPRB)    :: ZGM(KFLD_G,0:RALD%NMSMAX)
+REAL(KIND=JPHOOK) :: ZHOOK_HANDLE
+REAL(KIND=JPHOOK) :: ZHOOK_HANDLE1
```

Comment:
If you want to have more than one Dr Hook region in a code block, I don't think it's necessary to define a separate handle - you can just reuse `ZHOOK_HANDLE`. See e.g. [`LEDIR`](https://github.com/ecmwf-ifs/ectrans/blob/develop/src/trans/cpu/internal/ledir_mod.F90).

Reply:
Ah, didn't know that. Thanks!